webhook/webhook.go

```diff
@@ -529,7 +529,15 @@ func (ac *AdmissionController) mutate(ctx context.Context, kind metav1.GroupVers
 
 	var patches []jsonpatch.JsonPatchOperation
 
-	err := updateGeneration(ctx, &patches, oldObj, newObj)
+	// Add these before defaulting fields, otherwise defaulting may cause an illegal patch because
+	// it expects the round tripped through Golang fields to be present already.
+	roundTripPatch, err := roundTripPatch(newBytes, newObj)
```

Comment:
the variable name is not really good, since it shadows the function name at that point.

Reply:
Done.